### PR TITLE
Spring boot upgrade 2.1.2.RELEASE

### PR DESCRIPTION
**Pull request by AbhayChandel**
This PR is for issue https://github.com/devonfw/devon4j/issues/12

### CODE REVIEW

File: boms/bom/pom.xml [hunk 1]
```diff
@@ -88,19 +88,10 @@
...
       <dependency>
-        <groupId>org.hibernate.javax.persistence</groupId>
-        <artifactId>hibernate-jpa-2.1-api</artifactId>
-        <version>1.0.0.Final</version>
-      </dependency>
-      <!--
-        <dependency>
         <groupId>javax.persistence</groupId>
         <artifactId>javax.persistence-api</artifactId>
-        <version>2.2</version>
```
Comment (hohwille) [inline boms/bom/pom.xml]: why did you remove the version? IMHO this is a mistake. A dependency in depencenyManagement without a version seems pointless to me. Or is this a miss-interpretation of the diffs?
https://github.com/devonfw/devon4j/blob/d9d7403493513a24a14b363d3701fbd67b6d72ce/boms/bom/pom.xml#L93

Comment (hohwille) [reply boms/bom/pom.xml]: If that dependency comes in 2.2+ version via springs BOMs and we do not want to repeat it here, you should remove it entirely.

File: modules/batch/src/main/java/com/devonfw/module/batch/common/base/SpringBootBatchCommandLine.java [hunk 2]
```diff
@@ -172,7 +173,7 @@ public void execute(Operation operation, List<String> configurations, String job
     SpringApplication app = new SpringApplication(configurationClasses);
 
     // no (web) server needed
-    app.setWebEnvironment(false);
+    app.setWebApplicationType(WebApplicationType.NONE);
```
Comment (hohwille) [inline modules/batch/src/main/java/com/devonfw/module/batch/common/base/SpringBootBatchCommandLine.java]: Fine fix. However, this reminds me of the fact that our batch still needs rework and polising as this is old fashion style code. But removing the spring XMLs and moving to a modern spring-boot only config is not in the scope of this PR, so all perfectly right.

File: boms/bom/pom.xml (hunk 1 above)
Comment (AbhayChandel) [reply boms/bom/pom.xml]: Nice catch @hohwille , javax.persistence-api:2.2 is already defined in spring-boot-depenedencies:2.1.2.RELEASE. So I think it is better to remove
it completely from devon4j-bom. Removed and changes pushed to the PR.

**Comment by AbhayChandel** [comment]
> @AbhayChandel Thanks. Looking fine now. As we rely on an external dependencyManagement then, can you confirm this:
> 
> * we do not depend on this proprietary dependency anymore `org.hibernate.javax.persistence:hibernate-jpa-2.1-api`?
> * instead we depend on `javax.persistence:javax.persistence-api:2.2`, correct?

Yes, we are no more depending devon4j on `org.hibernate.javax.persistence:hibernate-jpa-2.1-api?`. It has been replaced with `javax.persistence:javax.persistence-api:2.2`.
